src/LndGrpc/Client.hs

```diff
@@ -80,11 +84,22 @@ runStreamServer rpc env req handler = do
     Right (Left e) ->
       Left $ LndGrpcError e
     Left e ->
-      Left . LndGrpcException $ inspect e
+      Left e
   where
     this =
       runClientIO $
         bracket
           (setupGrpcClient $ envLndConfig env)
           close
           (\grpc -> rawStreamServer rpc grpc () req $ const handler)
+
+catchExceptions :: MonadUnliftIO m => m (Either LndError a) -> m (Either LndError a)
+catchExceptions x =
+  x
+    `catches` [ Handler (\(e :: BlockedIndefinitelyOnMVar) -> pure (Left $ LndGrpcException $ inspect e)),
+                Handler
+                  ( \(ex :: IOException) -> case stripPrefix ("Network.Socket.connect" :: Text) (pack $ Prelude.show ex) of
```

Comment:
Should it be IOException? Maybe it's too wide + you need to parse text which is not ideal. Test this, maybe such handler will be enough:

https://github.com/21it/src/blob/a22ac3f061d88025ef6c56a6e47f5b3ec37d8228/bitfinex-client/src/BitfinexClient/Rpc/Generic.hs#L138-L151

Reply:
It catches IOException with type NoSuchThing, Looks like HttpException is not thrown by runUnary and cannot be catched.

Reply:
Done